src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -44,56 +44,58 @@ public static class PackageDownloader
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            // While CancellationTokenSource has a dispose, if we call it before the tasks using
+            // the token are done we risk having these tasks fail with a object disposed exception so
+            // this should not be in a using unless we want to block until all tasks are done/cancled
+            var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```

Comment:
@deepakaravindr 
The package is found in the cache so we trigger the task.Any with completed status of success.  The timing issue here is that for another task we are in the PluginCredentailProvider execute before registering the kill on the cancellation token.  According to https://msdn.microsoft.com/en-us/library/dd321635(v=vs.110).aspx, this is an expected exception in this case and it is the exception we were observing.

From the various blogs I have seen there is a lot of uncertainty around the necessity of disposing of cancelation tokens.  In general it seems to be considered safer to not dispose them.

We did evaluate the option to put a block in this method until all tasks completed so that exiting the using could safely dispose of the token.  This however would block progress of the command until all tasks were done and it seemed clear by the code that the desire here was to finish when the first task succeeded without waiting to see what the other tasks were doing.